MTD geometry: update BTL description

fabiocos opened the pull request:
#### PR description:

A new implementation of the BTL geometry is proposed, and all the software components related to it are updated accordingly. The model includes now a much more detailed description of all the passive components, and the material budget validation code is updated to monitor this new scenario. The crystal layout is conceptually unchanged, but the numbering is revised to better reflect the real engineering structure of the detector. In particular at tracking level the models move from an artificial definition of 3 x 16 crystal matrices to simple arrays of 16 crystals, increasing by a factor of 3 the number of modules. A detailed description of the various updates can be found in

[https://indico.cern.ch/event/1183866/contributions/4973544/attachments/2484059/4264823/btl_geometry.pdf](https://indico.cern.ch/event/1183866/contributions/4973544/attachments/2484059/4264823/btl_geometry.pdf)
[https://indico.cern.ch/event/1183866/contributions/4973544/attachments/2484059/4264824/MTDDPG_BTLgeometryV2_20220722.pdf](https://indico.cern.ch/event/1183866/contributions/4973544/attachments/2484059/4264824/MTDDPG_BTLgeometryV2_20220722.pdf)
[https://indico.cern.ch/event/1205335/contributions/5068947/attachments/2519146/4331505/MTDDPG_BTLgeometryV2_20220930.pdf](https://indico.cern.ch/event/1205335/contributions/5068947/attachments/2519146/4331505/MTDDPG_BTLgeometryV2_20220930.pdf)

This new BTL model requires the Outer Tracker boundary to be moved, and therefore needs to be based on tracker T31 or later, as implemented in #39592 .
Two new geometry scenarios have been proposed, based on earlier scenario and updating MTD (and Tracker), namely
D95 (from D88) and D96 (from D92).

#### PR validation:

All the standalone geometry tests (including unit tests) provide meaningful results. No overlap is detected by using Geant4 checks. RelVal worflows have been added for scenarios D95 and D96, they run without issues, and inspection of a 10k sample of SingleMuonPt100 histograms shows meaningful results. 

The test workflows will need to be updated as soon as #39571 is integrated.

fabiocos committed:
New BTL geometry by M. Casarsa, BTL numbering scheme, structure topology and sensitive detector definition update, adapt GeometricTiminbgDet geometry and DetLayers builds
Rebased onto CMSSW_12_5_X_2022-07-18-2300

fabiocos committed:
swap y and z coordinates for crystals and modules, fix rectangular topology parameters, add units to BTL translations, update BTLModule dimensions to restore functionality of RectangularMTDTopology, fix handeness of ReadoutUnit rotation, add material budget analysis for new scenario, added LYSO-Ce material, added detailed description of the tracker-BTL interface and adjusted the rails position

fabiocos committed:
Add new scenarios D95 and D96 for MTD I16, new BTL geometry

fabiocos committed:
Add relval workflows for geometry scenarios D95 and D96

fabiocos commented:
@casarsa FYI

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-39670/32473

- This PR adds an extra 396KB to repository

- Found files with invalid states:
  - Geometry/CMSCommonData/python/cmsExtendedGeometry2026D99XML_cfi.py:
    - Added: 0d17d6e821aaa824107a822df429ed6dd6b8cec1
    - Deleted: 01b8abc4ad7293e64a9dd10e16a0fcc0e0c1aeff
  - Geometry/CMSCommonData/data/dd4hep/cmsExtendedGeometry2026D99.xml:
    - Added: 0d17d6e821aaa824107a822df429ed6dd6b8cec1
    - Deleted: 01b8abc4ad7293e64a9dd10e16a0fcc0e0c1aeff
  - Configuration/Geometry/python/GeometryExtended2026D99_cff.py:
    - Added: 0d17d6e821aaa824107a822df429ed6dd6b8cec1
    - Deleted: 01b8abc4ad7293e64a9dd10e16a0fcc0e0c1aeff
  - Configuration/Geometry/python/GeometryExtended2026D99Reco_cff.py:
    - Added: 0d17d6e821aaa824107a822df429ed6dd6b8cec1
    - Deleted: 01b8abc4ad7293e64a9dd10e16a0fcc0e0c1aeff
  - Configuration/Geometry/python/GeometryDD4hepExtended2026D99_cff.py:
    - Added: 0d17d6e821aaa824107a822df429ed6dd6b8cec1
    - Deleted: 01b8abc4ad7293e64a9dd10e16a0fcc0e0c1aeff
  - Configuration/Geometry/python/GeometryDD4hepExtended2026D99Reco_cff.py:
    - Added: 0d17d6e821aaa824107a822df429ed6dd6b8cec1
    - Deleted: 01b8abc4ad7293e64a9dd10e16a0fcc0e0c1aeff

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/Geometry/README.md  modified in PR(s): #39571
  - File  Configuration/PyReleaseValidation/python/relval_2026.py  modified in PR(s): #39571
  - File  Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py  modified in PR(s): #39571, #33532, #39450

Code check has found code style and quality issues which could be resolved by applying following patch(s)

- **code-checks**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-39670/32473/code-checks.patch
e.g. `curl -k https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-39670/32473/code-checks.patch | patch -p1`
You can also run `scram build code-checks` to apply code checks directly

fabiocos committed:
scram b code-checks

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-39670/32474

- This PR adds an extra 396KB to repository

- Found files with invalid states:
  - Geometry/CMSCommonData/python/cmsExtendedGeometry2026D99XML_cfi.py:
    - Added: 0d17d6e821aaa824107a822df429ed6dd6b8cec1
    - Deleted: 01b8abc4ad7293e64a9dd10e16a0fcc0e0c1aeff
  - Geometry/CMSCommonData/data/dd4hep/cmsExtendedGeometry2026D99.xml:
    - Added: 0d17d6e821aaa824107a822df429ed6dd6b8cec1
    - Deleted: 01b8abc4ad7293e64a9dd10e16a0fcc0e0c1aeff
  - Configuration/Geometry/python/GeometryExtended2026D99_cff.py:
    - Added: 0d17d6e821aaa824107a822df429ed6dd6b8cec1
    - Deleted: 01b8abc4ad7293e64a9dd10e16a0fcc0e0c1aeff
  - Configuration/Geometry/python/GeometryExtended2026D99Reco_cff.py:
    - Added: 0d17d6e821aaa824107a822df429ed6dd6b8cec1
    - Deleted: 01b8abc4ad7293e64a9dd10e16a0fcc0e0c1aeff
  - Configuration/Geometry/python/GeometryDD4hepExtended2026D99_cff.py:
    - Added: 0d17d6e821aaa824107a822df429ed6dd6b8cec1
    - Deleted: 01b8abc4ad7293e64a9dd10e16a0fcc0e0c1aeff
  - Configuration/Geometry/python/GeometryDD4hepExtended2026D99Reco_cff.py:
    - Added: 0d17d6e821aaa824107a822df429ed6dd6b8cec1
    - Deleted: 01b8abc4ad7293e64a9dd10e16a0fcc0e0c1aeff

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/Geometry/README.md  modified in PR(s): #39571
  - File  Configuration/PyReleaseValidation/python/relval_2026.py  modified in PR(s): #39571
  - File  Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py  modified in PR(s): #39571, #33532, #39450

cmsbuild commented:
A new Pull Request was created by  @fabiocos (Fabio Cossutti) for master.

It involves the following packages:

- Configuration/Geometry (**geometry, upgrade**)
- Configuration/PyReleaseValidation (**pdmv, upgrade**)
- Configuration/StandardSequences (**operations**)
- DataFormats/ForwardDetId (**upgrade, simulation**)
- Geometry/CMSCommonData (**geometry, upgrade**)
- Geometry/MTDCommonData (**geometry, upgrade**)
- Geometry/MTDGeometryBuilder (**geometry, upgrade**)
- Geometry/MTDNumberingBuilder (**geometry, upgrade**)
- Geometry/MTDSimData (**geometry, upgrade**)
- RecoLocalFastTime/FTLCommonAlgos (**upgrade, reconstruction**)
- RecoMTD/DetLayers (**upgrade, reconstruction**)
- Validation/Geometry (**geometry, dqm**)


@rappoccio, @syuvivida, @bbilin, @ianna, @bsunanda, @perrotta, @civanch, @makortel, @ahmad3213, @cmsbuild, @mandrenguyen, @davidlange6, @Dr15Jones, @emanueleusai, @mdhildreth, @AdrianoDee, @jfernan2, @kskovpen, @micsucmed, @fabiocos, @clacaputo, @srimanob, @sunilUIET, @pmandrik, @rvenditti can you please review it and eventually sign? Thanks.
@VourMa, @felicepantaleo, @kpedro88, @Martin-Grunewald, @bsunanda, @trtomei, @rishabhCMS, @mmusich, @slomeo, @youyingli, @vargasa, @makortel, @JanFSchulte, @missirol, @beaucero, @GiacomoSguazzoni, @rovere, @VinInn, @ebrondol, @mtosi, @dgulhan, @rbartek, @apsallid, @lecriste this is something you requested to watch as well.
@perrotta, @dpiparo, @rappoccio you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

fabiocos commented:
please test workflow 42234.0,42634.0

fabiocos commented:
I may squash a subset of commits later on to eliminate the unwanted files, although they are not particularly large, since there is not a special value in the detailed history

fabiocos commented:
The failure in the ```RecoMTD/DetLayers``` unit test looks like a trivial formatting problem induced by the additional information I have put in the printout. I will update the reference to prevent it.

cmsbuild commented:
-1

**Failed Tests**: UnitTests 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-405ec8/28102/summary.html
**COMMIT**: 8ad0484610d07ed59b3f68b176fb9715fa7d1ddb
**CMSSW**: CMSSW_12_6_X_2022-10-06-2300/el8_amd64_gcc10
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/39670/28102/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Unit Tests

I found errors in the following unit tests:

<pre>---> test RecoMTDDetLayersTestDriver had ERRORS
</pre>

## Comparison Summary



@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-405ec8/41834.0_TTbar_14TeV+2026D94+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-405ec8/42234.0_TTbar_14TeV+2026D95+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-405ec8/42634.0_TTbar_14TeV+2026D96+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal

**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 8 differences found in the comparisons
 - DQMHistoTests: Total files compared: 49
 - DQMHistoTests: Total histograms compared: 3391103
 - DQMHistoTests: Total failures: 6
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3391075
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 48 files compared)
 - Checked 204 log files, 49 edm output root files, 49 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_6_X_2022-10-06-2300+405ec8/53262/triggerResults): no differences found

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-39670/32481

- This PR adds an extra 164KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/Geometry/README.md  modified in PR(s): #39571
  - File  Configuration/PyReleaseValidation/python/relval_2026.py  modified in PR(s): #39571
  - File  Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py  modified in PR(s): #39571, #33532, #39450

cmsbuild commented:
Pull request #39670 was updated. @rappoccio, @syuvivida, @bbilin, @ianna, @bsunanda, @perrotta, @civanch, @makortel, @ahmad3213, @cmsbuild, @mandrenguyen, @davidlange6, @Dr15Jones, @emanueleusai, @mdhildreth, @AdrianoDee, @jfernan2, @kskovpen, @micsucmed, @fabiocos, @clacaputo, @srimanob, @sunilUIET, @pmandrik, @rvenditti can you please check and sign again.

fabiocos commented:
please test workflow 42234.0,42634.0 with cms-data/Geometry-TestReference#11

cmsbuild commented:
-1

**Failed Tests**: UnitTests 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-405ec8/28110/summary.html
**COMMIT**: eba79de4adcada21a096e75cc88dc2f159c5a7a3
**CMSSW**: CMSSW_12_6_X_2022-10-07-1100/el8_amd64_gcc10
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/39670/28110/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Unit Tests

I found errors in the following unit tests:

<pre>---> test test-das-selected-lumis had ERRORS
</pre>

## Comparison Summary



@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-405ec8/41834.0_TTbar_14TeV+2026D94+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-405ec8/42234.0_TTbar_14TeV+2026D95+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-405ec8/42634.0_TTbar_14TeV+2026D96+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal

**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 4 differences found in the comparisons
 - DQMHistoTests: Total files compared: 49
 - DQMHistoTests: Total histograms compared: 3391103
 - DQMHistoTests: Total failures: 3
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3391078
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 48 files compared)
 - Checked 204 log files, 49 edm output root files, 49 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_6_X_2022-10-07-1100+405ec8/53269/triggerResults): no differences found

fabiocos commented:
Rebased on CMSSW_12_6_0_pre3.
The previous test failures looked unrelated to this PR

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-39670/32509

- This PR adds an extra 180KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/Geometry/README.md  modified in PR(s): #39571
  - File  Configuration/PyReleaseValidation/python/relval_2026.py  modified in PR(s): #39571
  - File  Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py  modified in PR(s): #39571, #33532, #39450

cmsbuild commented:
Pull request #39670 was updated. @rappoccio, @syuvivida, @bbilin, @ianna, @bsunanda, @perrotta, @civanch, @makortel, @ahmad3213, @cmsbuild, @mandrenguyen, @davidlange6, @Dr15Jones, @emanueleusai, @mdhildreth, @AdrianoDee, @jfernan2, @kskovpen, @micsucmed, @fabiocos, @clacaputo, @srimanob, @sunilUIET, @pmandrik, @rvenditti can you please check and sign again.

fabiocos commented:
please test workflow 42234.0,42634.0 with https://github.com/cms-data/Geometry-TestReference/pull/11

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-405ec8/28150/summary.html
**COMMIT**: f692a16913766fe7f4aa3e6f099ace1603927a0f
**CMSSW**: CMSSW_12_6_X_2022-10-09-2300/el8_amd64_gcc10
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/39670/28150/install.sh` to create a dev area with all the needed externals and cmssw changes.



The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - @makortel cms-sw/cmssw#39636
 - @cms-sw cms-sw/cmsdist#8123
 - @cms-sw cms-sw/cmsdist#8122

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-405ec8/28150/git-recent-commits.json
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-405ec8/28150/git-merge-result


## Comparison Summary



@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-405ec8/41834.0_TTbar_14TeV+2026D94+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-405ec8/42234.0_TTbar_14TeV+2026D95+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-405ec8/42634.0_TTbar_14TeV+2026D96+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal

**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 8 differences found in the comparisons
 - DQMHistoTests: Total files compared: 49
 - DQMHistoTests: Total histograms compared: 3392309
 - DQMHistoTests: Total failures: 9
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3392278
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 48 files compared)
 - Checked 204 log files, 49 edm output root files, 49 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_6_X_2022-10-09-2300+405ec8/53311/triggerResults): no differences found

emanueleusai commented:
+1

mandrenguyen commented:
+1

srimanob commented:
Hi @fabiocos 
Please take note on https://github.com/cms-sw/cmssw/pull/39571 which the workflow id will be reset. I can ping to this PR again when it merges.

fabiocos commented:
@srimanob thanks, at that point I believe this PR will be just not mergeable, and I will need anyway to rebase

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-39670/32533

- This PR adds an extra 200KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py  modified in PR(s): #33532, #39450

cmsbuild commented:
Pull request #39670 was updated. @rappoccio, @syuvivida, @bbilin, @ianna, @bsunanda, @perrotta, @civanch, @makortel, @ahmad3213, @cmsbuild, @mandrenguyen, @davidlange6, @Dr15Jones, @emanueleusai, @mdhildreth, @AdrianoDee, @jfernan2, @kskovpen, @micsucmed, @fabiocos, @clacaputo, @srimanob, @sunilUIET, @pmandrik, @rvenditti can you please check and sign again.

fabiocos committed:
Add renumbered matrix workflows for scenarios D95 and D96

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-39670/32534

- This PR adds an extra 200KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py  modified in PR(s): #33532, #39450

cmsbuild commented:
Pull request #39670 was updated. @rappoccio, @syuvivida, @bbilin, @ianna, @bsunanda, @perrotta, @civanch, @makortel, @ahmad3213, @cmsbuild, @mandrenguyen, @davidlange6, @Dr15Jones, @emanueleusai, @mdhildreth, @AdrianoDee, @jfernan2, @kskovpen, @micsucmed, @fabiocos, @clacaputo, @srimanob, @sunilUIET, @pmandrik, @rvenditti can you please check and sign again.

fabiocos commented:
please test workflow 42234.0,42634.0 with https://github.com/cms-data/Geometry-TestReference/pull/11

fabiocos commented:
please abort

test numbers have changed!

fabiocos commented:
please test workflow 23634.0,24034.0 with https://github.com/cms-data/Geometry-TestReference/pull/11

cmsbuild commented:
-1

**Failed Tests**: RelVals-INPUT 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-405ec8/28187/summary.html
**COMMIT**: d32ca36f138d9603565a077a3cd00604593a20da
**CMSSW**: CMSSW_12_6_X_2022-10-11-2300/el8_amd64_gcc10
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/39670/28187/install.sh` to create a dev area with all the needed externals and cmssw changes.



The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - @Dr15Jones cms-sw/cmssw#39637
 - @Dr15Jones cms-sw/cmssw#39626
 - @Dr15Jones cms-sw/cmssw#39632
 - @wddgit cms-sw/cmssw#39673

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-405ec8/28187/git-recent-commits.json
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-405ec8/28187/git-merge-result


## RelVals-INPUT

The relvals timed out after 4 hours.


## Comparison Summary



@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-405ec8/20834.0_TTbar_14TeV+2026D88+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-405ec8/20834.75_TTbar_14TeV+2026D88_HLT75e33+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HLT75e33+HARVESTGlobal
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-405ec8/20896.0_CloseByPGun_CE_E_Front_120um+2026D88+CE_E_Front_120um_GenSimHLBeamSpotHGCALCloseBy+DigiTrigger+RecoGlobal+HARVESTGlobal
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-405ec8/20900.0_CloseByPGun_CE_H_Coarse_Scint+2026D88+CE_H_Coarse_Scint_GenSimHLBeamSpotHGCALCloseBy+DigiTrigger+RecoGlobal+HARVESTGlobal
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-405ec8/21034.999_TTbar_14TeV+2026D88PU_PMXS1S2PR+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+PREMIX_PremixHLBeamSpot14PU+DigiTriggerPU+RecoGlobalPU+HARVESTGlobalPU
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-405ec8/23234.0_TTbar_14TeV+2026D94+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-405ec8/23634.0_TTbar_14TeV+2026D95+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-405ec8/24034.0_TTbar_14TeV+2026D96+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal

**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 4 differences found in the comparisons
 - DQMHistoTests: Total files compared: 49
 - DQMHistoTests: Total histograms compared: 3392309
 - DQMHistoTests: Total failures: 3
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3392284
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 48 files compared)
 - Checked 204 log files, 49 edm output root files, 49 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_6_X_2022-10-11-2300+405ec8/53350/triggerResults): no differences found

srimanob commented:
test parameters:
- workflows = 23634.0,24034.0
- pull_request = https://github.com/cms-data/Geometry-TestReference/pull/11

srimanob commented:
@cmsbuild please test

Just retrigger the test to make a clean PR test result.

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-405ec8/28216/summary.html
**COMMIT**: d32ca36f138d9603565a077a3cd00604593a20da
**CMSSW**: CMSSW_12_6_X_2022-10-12-1100/el8_amd64_gcc10
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/39670/28216/install.sh` to create a dev area with all the needed externals and cmssw changes.



The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - @wddgit cms-sw/cmssw#39700
 - @bsunanda cms-sw/cmssw#39671
 - @bsunanda cms-sw/cmssw#39676
 - @cecilecaillol cms-sw/cmssw#39704
 - @Dr15Jones cms-sw/cmssw#39643
 - @Moanwar cms-sw/cmssw#39519
 - @yeckang cms-sw/cmssw#39487
 - @cms-sw cms-sw/cmsdist#8126

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-405ec8/28216/git-recent-commits.json
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-405ec8/28216/git-merge-result


## Comparison Summary



@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-405ec8/20834.0_TTbar_14TeV+2026D88+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-405ec8/20834.75_TTbar_14TeV+2026D88_HLT75e33+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HLT75e33+HARVESTGlobal
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-405ec8/20896.0_CloseByPGun_CE_E_Front_120um+2026D88+CE_E_Front_120um_GenSimHLBeamSpotHGCALCloseBy+DigiTrigger+RecoGlobal+HARVESTGlobal
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-405ec8/20900.0_CloseByPGun_CE_H_Coarse_Scint+2026D88+CE_H_Coarse_Scint_GenSimHLBeamSpotHGCALCloseBy+DigiTrigger+RecoGlobal+HARVESTGlobal
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-405ec8/21034.999_TTbar_14TeV+2026D88PU_PMXS1S2PR+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+PREMIX_PremixHLBeamSpot14PU+DigiTriggerPU+RecoGlobalPU+HARVESTGlobalPU
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-405ec8/23234.0_TTbar_14TeV+2026D94+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-405ec8/23634.0_TTbar_14TeV+2026D95+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-405ec8/24034.0_TTbar_14TeV+2026D96+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal

**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 15 differences found in the comparisons
 - DQMHistoTests: Total files compared: 49
 - DQMHistoTests: Total histograms compared: 3392309
 - DQMHistoTests: Total failures: 9
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3392278
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: -0.016 KiB( 48 files compared)
 - DQMHistoSizes: changed ( 138.5 ): -0.008 KiB       MessageLogger/Errors
 - DQMHistoSizes: changed ( 138.5 ): -0.008 KiB       MessageLogger/Warnings
 - Checked 204 log files, 49 edm output root files, 49 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_6_X_2022-10-12-1100+405ec8/53363/triggerResults): no differences found

srimanob commented:
Hi @cms-sw/reconstruction-l2 
Do you plan to update the comparison matrix, to reflect the change of workflow ID? Thanks.

slava77 commented:
> Do you plan to update the comparison matrix, to reflect the change of workflow ID? Thanks.

what is the status of 2026->2029 renaming PR?
If it's imminent, the matrix update will be needed again

srimanob commented:
> > Do you plan to update the comparison matrix, to reflect the change of workflow ID? Thanks.
> 
> what is the status of 2026->2029 renaming PR? If it's imminent, the matrix update will be needed again

Hi @slava77 
I think it is not urgent for that PR. We still have some in the pipeline and few new geometries to make. Can we postpone the update of year to 13_0_0_pre1?

fabiocos commented:
@srimanob I did a private comparison with muon and pion particle guns, and it looked reasonable.

slava77 commented:
> I think it is not urgent for that PR. We still have some in the pipeline and few new geometries to make. Can we postpone the update of year to 13_0_0_pre1?

OK

fabiocos commented:
@cms-sw/reconstruction-l2 @cms-sw/simulation-l2 @cms-sw/geometry-l2 @cms-sw/dqm-l2 @cms-sw/pdmv-l2 @cms-sw/upgrade-l2 do you have comments on the PR? Or can we proceed with its integration?

mandrenguyen commented:
@fabiocos 
Just to verify, you don't expect to see any differences in the comparisons? 
From the description, it seems that this is mostly a change to the implementation (and addition of new scenarios).
But do you confirm that this update to the baseline MTD geometry is physics invariant?

fabiocos commented:
@mandrenguyen this is indeed the target, we need to maintain the old geometries unchanged and data formats backward compatible. As you may see in ```BTLDetId``` there are additions of methods, but the existing interfaces are unchanged. And the update of constants and a few methods leaves the final results on old geometries unchanged, as also confirmed by the various unit tests. From the reconstruction point of view the most interesting update is in the ```DetLayers ``` geometry, but again for the new version, the old one provides the same output. I needed to update the reference because I added a few more items in the output that were not present earlier, but are useful to monitor now. But you may easily verify that the old items are the same. In the old geometry we had not the concept of readout unit number, and modules were organized in a different way.

fabiocos commented:
BTW, compared to previous partial approval, I just rebased on latest releases and updated the workflow numbering accordingly, but the geometry and reconstruction part is unchanged, as you may easily verify.

emanueleusai commented:
+1

mandrenguyen commented:
+1

srimanob commented:
+Upgrade

bsunanda commented:
+geometry

civanch commented:
+1

kskovpen commented:
+pdmv

srimanob commented:
@cmsbuild please test

Refresh the test. 
@perrotta @rappoccio Could you please consider to merge the PR after? We still have new geometry in the pile line to be added. Thx.

cmsbuild commented:
-1

**Failed Tests**: RelVals-INPUT 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-405ec8/28454/summary.html
**COMMIT**: d32ca36f138d9603565a077a3cd00604593a20da
**CMSSW**: CMSSW_12_6_X_2022-10-23-2300/el8_amd64_gcc10
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/39670/28454/install.sh` to create a dev area with all the needed externals and cmssw changes.


## RelVals-INPUT

The relvals timed out after 4 hours.


## Comparison Summary



@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-405ec8/23634.0_TTbar_14TeV+2026D95+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-405ec8/24034.0_TTbar_14TeV+2026D96+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal

**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 2 differences found in the comparisons
 - Reco comparison had  6 failed jobs
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3378384
 - DQMHistoTests: Total failures: 6
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3378356
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 47 files compared)
 - Checked 201 log files, 48 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_6_X_2022-10-23-2300+405ec8/53617/triggerResults): no differences found

fabiocos commented:
please test

perrotta commented on RecoMTD/DetLayers/src/MTDDetLayerGeometry.cc:
```diff
@@ -28,18 +28,17 @@ MTDDetLayerGeometry::MTDDetLayerGeometry() {}
-    }
-  } else {
+  if (geo == nullptr) {
     LogWarning("MTDDetLayers") << "No MTD geometry is available.";
```
As pointed out by the static analyzer: if either `geo` or `topo` is a `nullptr` the following builder methods will dereference a null pointer: should you more gracefully exit here, or do you rely on LogWarning + crash to identify it in the logs?

fabiocos replied:
@perrotta in principle you are correct, I might add an exception to let the code exit in a less brutal way. In practice this is such an extreme case that means something was seriously misconfigured in the job and this code should never run.
Given the fact that this issue is there since a while and this review is going already quite long, I would prefer to add such a protection to make the static analyzer happy in a send small dedicated PR, if you agree

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-405ec8/28464/summary.html
**COMMIT**: d32ca36f138d9603565a077a3cd00604593a20da
**CMSSW**: CMSSW_12_6_X_2022-10-23-2300/el8_amd64_gcc10
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/39670/28464/install.sh` to create a dev area with all the needed externals and cmssw changes.



The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - @bsunanda cms-sw/cmssw#39816
 - @swagata87 cms-sw/cmssw#39820
 - @cecilecaillol cms-sw/cmssw#39422
 - @belforte cms-sw/cmsdist#8135

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-405ec8/28464/git-recent-commits.json
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-405ec8/28464/git-merge-result


## Comparison Summary



@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-405ec8/23634.0_TTbar_14TeV+2026D95+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-405ec8/24034.0_TTbar_14TeV+2026D96+TTbar_14TeV_TuneCP5_GenSimHLBeamSpot14+DigiTrigger+RecoGlobal+HARVESTGlobal

**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 7 differences found in the comparisons
 - Reco comparison had  6 failed jobs
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3378384
 - DQMHistoTests: Total failures: 12
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3378350
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 47 files compared)
 - Checked 201 log files, 48 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_6_X_2022-10-23-2300+405ec8/53624/triggerResults): found differences in 1 / 46 workflows

perrotta commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will be automatically merged.

fabiocos commented:
@perrotta thanks, anyway please recall to merge also https://github.com/cms-data/Geometry-TestReference/pull/11 otherwise one unit test for ```RecoMTD/DetLayers``` will fail

fabiocos commented:
@perrotta I see a failure in the IB related to this PR in wf 20834.911 , I am a bit surprised because the DD4hep unit tests are all passing, nevertheless I need to investigate

srimanob commented:
```
----- Begin Fatal Exception 25-Oct-2022 06:40:34 CEST-----------------------
An exception of category 'BtlSimHitsValidation' occurred while
   [0] Processing  Event run: 1 lumi: 1 event: 3 stream: 2
   [1] Running path 'validation_step10'
   [2] Calling method for module BtlSimHitsValidation/'btlSimHitsValid'
Exception Message:
GeographicalID: 6280c000 (0) is invalid!
----- End Fatal Exception -------------------------------------------------
```

perrotta commented:
@fabiocos indeed that seems to be related: please have a look and let us know

fabiocos commented:
yes, this is the issue, the point is that for scenario D88 the detIds should not change, and indeed the units tests dumping the geometry both from DDD and DD4hep are passing. Looking into it...
